pkg/storage/pod.go

```diff
@@ -117,6 +119,16 @@ func (s *podStorage) Store(newPods []PodMetricsPoint) {
 								"timestamp", newContainer.Timestamp)
 						}
 					}
+				} else if lastContainer, found := lastPod[newContainer.Name]; found && newContainer.StartTime.After(lastContainer.Timestamp) {
```

Comment:
Have you considered moving those ifs before checking last container?
You don't need to check lastContainer to knew that container is fresh.

Reply:
All we use the condition `newContainer.StartTime.Before(newContainer.Timestamp) && newContainer.Timestamp.Sub(newContainer.StartTime) < s.scrapeDuration`?

but I think the period s.scrapeDuration maybe too long?

Reply:
What do you mean? We scrape metrics every scrapeDuration, so why do you think it maybe too long?

Reply:
ok，I see. 
Let me try to modify.